File: sylee723/3주차/PGS_두_큐_합_같게_만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        Queue<Integer> q1 = new ArrayDeque<>();
        Queue<Integer> q2 = new ArrayDeque<>();
        long sum1 = 0;
        long sum2 = 0;
        
        int len = queue1.length;
        
        for (int i = 0; i < len; i++) {
            q1.add(queue1[i]);
            sum1 += queue1[i];
        }
        for (int i = 0; i < len; i++) {
            q2.add(queue2[i]);
            sum2 += queue2[i];
        }
        
        int num;
        int answer = 0;
        while (sum1 != sum2) {
            if (answer > len * 4) {
                answer = -1;
                break;
            }
            
            if (sum1 > sum2) {
                num = q1.poll();
                sum1 -= num;
                sum2 += num;
                q2.add(num); 
            } else if (sum1 < sum2) {
                num = q2.poll();
                sum1 += num;
                sum2 -= num;
                q1.add(num);
            }
            answer++;
        }
        
        return answer;
    }
}

File: daehwa_park/3주차/PGS_두_큐_합_같게_만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = -1
    
    q1, q2 = deque(queue1), deque(queue2)
    s1, s2 = sum(queue1), sum(queue2)
    max_count = len(queue1) * 3
    
    if (sum(queue1) + sum(queue2)) % 2 == 1:
        return -1
    
    for i in range(max_count):
        if s1 > s2:
            num = q1.popleft()
            q2.append(num)
            s1 -= num
            s2 += num
        elif s2 > s1:
            num = q2.popleft()
            q1.append(num)
            s2 -= num
            s1 += num
        else:
            return i
        
    
    return answer